### PR TITLE
Gradle Recipe AddDependency without version

### PR DESCRIPTION
**Pull request by nicolasb29**
For [#3208](https://github.com/openrewrite/rewrite/issues/3208)

### CODE REVIEW

**Review by shanman190 — CHANGES_REQUESTED** [review]
@nicolasb29, thanks for your contribution! I've added a couple of minor comments.

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependency.java [hunk 1]
```diff
@@ -53,7 +53,8 @@ public class AddDependency extends Recipe {
 
     @Option(displayName = "Version",
             description = "An exact version number or node-style semver selector used to select the version number.",
-            example = "29.X")
+            example = "29.X",
+            required = false)
     String version;
```
[the remark this answers is not in the document]
Comment (nicolasb29) [reply rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependency.java:59]: @shanman190 I just added the annotation ```@Nullable``` on the version field.

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependencyVisitor.java [hunk 2]
```diff
@@ -113,12 +115,12 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, Execu
...
-                               "\n}";
+                        configuration + " \"" + groupId + ":" + artifactId + (version == null ? "" : ":" + version) + (version == null || classifier == null ? "" : ":" + classifier) + (extension == null ? "" : "@" + extension) + "\"" +
+                        "\n}";
             } else {
                 codeTemplate = "dependencies {\n" +
-                               configuration + " group: \"" + groupId + "\", name: \"" + artifactId + "\", version: \"" + version + "\"" + (classifier == null ? "" : ", classifier: \"" + classifier + "\"") + (extension == null ? "" : ", ext: \"" + extension + "\"") +
-                               "\n}";
+                        configuration + " group: \"" + groupId + "\", name: \"" + artifactId + "\"" + (version == null ? "" :  ", version: \"" + version + "\"") + (version == null || classifier == null ? "" : ", classifier: \"" + classifier + "\"") + (extension == null ? "" : ", ext: \"" + extension + "\"") +
```
[the remark this answers is not in the document]
Comment (nicolasb29) [reply rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependencyVisitor.java]: @shanman190 I modified the map notation to keep classifier even if version is not present. I also added some tests.

**Review by shanman190 — APPROVED** [review]
LGTM! Thank you for the contribution, @nicolasb29!

**Pull request merged**